### PR TITLE
Slack integration, correcting typos and small refactor

### PR DESCRIPTION
**Pull request by wefner**

### CODE REVIEW

File: notifierlib/channels/slack.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                 reply_broadcast=False):
+        super(Slack, self).__init__(name)
+        self.private = private
+        self.channel = self.__determine_channel(channel)
+        self.template = template
+        self.site = 'https://slack.com/api'
+        self.reply_broadcast = reply_broadcast
+        self.__token = token
+
+    def __determine_channel(self, channel):
```
Comment (costastf) [inline notifierlib/channels/slack.py]: This seems very overkill for what it does.
I think that line 38 could just be 
```self.channel = channel if self.private else '#{}'.format(channel)```

File: notifierlib/channels/slack.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            return False
+
+        arguments = {'channel': self.channel,
+                     'token': self.__token,
+                     'text': body,
+                     'reply_broadcast': self.reply_broadcast,
+                     'as_user': response.json().get('user_id')}
+        response = requests.post(url='{site}/chat.postMessage'.format(site=self.site),
+                                 data=arguments)
+        self._logger.debug(('Message sent successfully. Response text: '
```
Comment (costastf) [inline notifierlib/channels/slack.py]: This is a bit optimistic. Just the fact that the request did not raise an exception does not mean that the message was actually sent successfully. I think we should check the response and act accordingly.

**Review by costastf — COMMENTED** [review]
Small comments on the code.

**Comment by wefner** [comment]
Thanks! Comments are addressed

File: notifierlib/channels/slack.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                     'token': self.__token,
+                     'text': body,
+                     'reply_broadcast': self.reply_broadcast,
+                     'as_user': response.json().get('user_id')}
+        response = requests.post(url='{site}/chat.postMessage'.format(site=self.site),
+                                 data=arguments)
+        if response.json().get('ok'):
+            self._logger.debug(('Message sent successfully. Response text: '
+                                '{response}').format(response=response.text))
+        return True
```
Comment (costastf) [inline notifierlib/channels/slack.py]: Wouldn't it be wiser to return the response.ok status here instead of true, blindly?

**Comment by costastf** [comment]
Another very small comment on the return

**Comment by wefner** [comment]
Updated response. I didn't want to return `response.ok` since it will always be `200` and what it matters is the `json` response in its content. So I have this `result` "flag". What do you think?

**Review by costastf — APPROVED** [review]
We should move the logic of the authentication and such out of the notify for brevity. Looks great, thanks!

**Pull request merged**